### DISCUSSION TITLE
How to change the base path?

### DISCUSSION THREAD

**Discussion by danielniccoli** [litestar-org/litestar#2964] [Q&A] [answered]
> `Litestar` is the root level of the app - it has the base path of `/` and all root level Controllers, Routers and Route Handlers should be registered on it.

https://docs.litestar.dev/2/reference/app.html#litestar.app.Litestar

How can I change the base path from `/` to `/api/`? This should not affect `/schema`. I would like to avoid adding `/api/` to every handler.

**Comment by provinzkraut** [comment, marked as answer]
Just add your route handlers to a `Router` instead of to `Litestar` directly. Then you can change the base path there for everything except the `/schema`:

```python
api_router = Router(path="/api", route_handlers=...)
app = Litestar(route_handlers=[api_router])
```

**Reply by danielniccoli** [reply]
Thank you!

**Comment by peterschutt** [comment]
The litestar app object accepts [`path`][1] as of 2.8.

[1]: https://docs.litestar.dev/latest/reference/app.html#litestar.app.Litestar.params.path
